detect_secrets/plugins/common/filters.py

```diff
@@ -1,35 +1,75 @@
 """
-Heuristic, false positive filters that are shared across all plugin types.
+False positive heuristic filters that are shared across all plugin types.
 This abstraction allows for development of later ML work, or further
 heuristical determinations (e.g. word filter, entropy comparator).
 """
 import string
 
+from detect_secrets.util import is_python_2
 
-def is_false_positive(secret):
-    for func in [
-        is_sequential_string,
-    ]:
-        if func(secret):
-            return True
 
-    return False
+def is_false_positive(secret, automaton):
+    """
+    :type secret: str
 
+    :type automaton: ahocorasick.Automaton|None
+    :param automaton: optional automaton for ignoring certain words.
 
-def is_sequential_string(secret):
+    :rtype: bool
+    Returns True if any false positive heuristic function returns True.
     """
-    Returns true if string is sequential.
+    return any(
+        func(secret, automaton)
+        for func in
+        (
+            _is_found_with_aho_corasick,
+            _is_sequential_string,
+        )
+    )
+
+
+def _is_found_with_aho_corasick(secret, automaton):
+    """
+    :type secret: str
+
+    :type automaton: ahocorasick.Automaton|None
+    :param automaton: optional automaton for ignoring certain words.
+
+    :rtype: bool
+    Returns True if secret contains a word in the automaton.
+    """
+    if not automaton:
+        return False
+
+    if is_python_2():  # pragma: no cover
+        # Due to pyahocorasick
+        secret = secret.encode('utf-8')
+
+    try:
+        # .lower() to make everything case-insensitive
+        next(automaton.iter(string=secret.lower()))
```

Comment:
I feel like `.lower` here makes it so that you can search for where search the automata, and never miss that it's case insensitive.

Adding another `secret = secret.something()` is an extra line that, if we were `.lower()`ing more than once I'd be happy to add, but since it's one time only this seems okay.

Reply:
👍 makes sense